### PR TITLE
sample: Add documentation

### PR DESCRIPTION
**Pull request by tbarbette**
Show usage, how to generate certs, etc. We had students bumping on this so...

### CODE REVIEW

File: sample/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Example
+-------
+
+Generate the certificates:
+```
+openssl req -x509 -newkey rsa:4096 -days 365 -keyout ca-key.pem -out ca-cert.pem
+openssl req -newkey rsa:4096 -keyout server-key.pem -out server-req.pem
+```
+These commands will prompt a few questions, you don't need to put actual data
```
Comment (huitema) [inline sample/README.md:26]: RSA 4096 will make the initial exchange kind of slow. Using ECDSA is a bit more complex, so I get sticking with RSA for the sample is fine, but 4096 is probably overkill. 2048 would be adequate.

File: sample/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+picoquic sample
+===============
+
+The sample program is a simple QUIC client/server demo.
+
+Building
+--------
+picoquic\_sample is built as part of the compilation process of picoquic. It is
+will be available in the root folder.
+
```
Comment (huitema) [inline sample/README.md:10]: Typo, "it is will"

File: sample/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+```
+mkdir server_files
+echo "Hello world!" >> ./server_files/index.htm
+```
+
+And run the server:
+```
+sudo ./picoquic_sample server 443 ./ca-cert.pem ./server-key.pem ./server_files
+```
+
```
Comment (huitema) [inline sample/README.md:39]: Or run on high port number if you don't want to use sudo. Most of the QUIC interop work was done on port 4433.

File: sample/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Then, test if you can reach it using the client:
+```
+sudo ./picoquic_sample client localhost 443 /tmp index.htm
+```
```
Comment (huitema) [inline sample/README.md:43]: You don't need sudo there.

**Comment by tbarbette** [comment]
Should be done ;) Thanks for the review!

**Pull request merged**